src/riot-rs-embassy/src/thread_executor.rs

```diff
@@ -0,0 +1,180 @@
+use core::marker::PhantomData;
+use embassy_executor::{raw, Spawner};
+use riot_rs_threads::{current_pid, thread_flags, thread_flags::ThreadFlags, ThreadId};
+
+const THREAD_FLAG_EXECUTOR_WAKEUP: ThreadFlags = 0x01;
+
+#[export_name = "__pender"]
+fn __pender(context: *mut ()) {
+    // Safety: `context` is a `ThreadId` passed by `ThreadExecutor::new`.
+    let thread_id = ThreadId::new(context as usize as u8);
+
+    thread_flags::set(thread_id, THREAD_FLAG_EXECUTOR_WAKEUP);
+}
+
+/// Thread mode executor for RIOT-rs threads
+///
+pub struct Executor {
+    inner: raw::Executor,
+    not_send: PhantomData<*mut ()>,
```

Comment:
Good question, I copied that from upstream ...

Reply:
Let's assume that we want to be explicit on this? While technically, `!Send` and `!Sync` are inherited, that might change, and then our `Executor` suddenly becomes `Send` or `Sync`.

Reply:
Not sure if I am a fan of that. I briefly looked into the embassy implementation and I believe they only have it for historic reasons.

But I don't feel strongly about it, so I am also okay with keeping it.

Reply:
Just to be clear, this executor instance cannot be moved between threads, b/c `new()` hard sets the thread id that `__pender()` is gonna use. So `run()` must be called on that same thread, our `Executor` cannot be `Send`. Isn't the `PhantomData<*mut ()>` *the* method to express this?

Reply:
> So `run()` must be called on that same thread, our `Executor` cannot be `Send`. Isn't the PhantomData<*mut ()> the method to express this?

Yes but that's already done internally by embassy. Their executor holds the `struct Pender(*mut ())` with the thread id info. I don't expect that this will ever change, so it's guaranteed that our `Executor` will never be `Send`.  

I don't find `PhantomData<*mut ()>` intuitive. But again, I don't feel strongly about it :)

Reply:
> Yes but that's already done internally by embassy. Their executor holds the struct Pender(*mut ()) with the thread id info. I don't expect that this will ever change, so it's guaranteed that our Executor will never be Send.

Actually, I just noticed they explicitly implement `Send` and `Sync` for the `Pender`, so the `raw::Executor` is only `!Send` and `!Sync`  because they also implement a `_not_sync: PhantomData<*mut ()>,` marker on it.

Let's keep it then. But could you please add a comment on it?

Reply:
Comment sounds very reasonable. :) Done.